core/dbt/parser/manifest.py

```diff
@@ -1439,6 +1440,63 @@ def _process_refs(
         node.depends_on.add_node(target_model_id)
 
 
+def _process_metric_node(
+    manifest: Manifest,
+    current_project: str,
+    metric: Metric,
+) -> None:
+    """Sets a metric's input_measures"""
+
+    # This ensures that if this metrics input_measures have already been set
+    # we skip the work. This could happen either due to recursion or if multiple
+    # metrics derive from another given metric.
+    # NOTE: This does not protect against infinite loops
+    if len(metric.type_params.input_measures) > 0:
+        return
+
+    if metric.type is MetricType.SIMPLE or metric.type is MetricType.CUMULATIVE:
+        assert (
+            metric.type_params.measure is not None
+        ), f"{metric} should have a measure defined, but it does not."
+        metric.type_params.input_measures.append(metric.type_params.measure)
+
+    elif metric.type is MetricType.DERIVED or metric.type is MetricType.RATIO:
+        input_metrics = metric.input_metrics
+        if metric.type is MetricType.RATIO:
+            if metric.type_params.numerator is None or metric.type_params.denominator is None:
+                raise dbt.exceptions.ParsingError(
```

Comment:
I wasn't wholly sure if this should be considered a `ParsingError` or `CompilationError` given the point at which we are in the manifest loading process. Happy to switch it up if a different error than what I went with is more appropriate.

Reply:
I'd call it a parsing error. The boundary isn't very clear because certain Jinja errors are always called compilation errors no matter whether they're in parsing or compilation.